docs/upgrading.md

```diff
@@ -5,6 +5,11 @@ the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summar
 
 ## Upgrading to v3.6
 
+### JSON templating fix
+
+When accessing a structure or array from json structure in a `jsonpath` expression you could end up getting a golang representation of the structure in error.
```

Comment:
While the test cases in #12909 are `jsonpath` specific, the code is not, as far as I understand, otherwise it could have been applied to the `jsonpath` helper specifically.

The wording is also duplicative and inconsistent:
```suggestion
When accessing a map or array from JSON in a `jsonpath` expression, you would get a Golang representation.
```
- consistently use "JSON", not both "json" and "JSON" (esp in the same section)
  - "Golang" should similarly be capitalized as a proper noun
- simplify, per k8s style guide
  - "structure" is not needed (was previously used 3 times in one sentence) and is ambiguous, whereas "map" is a data structure consistent with "array"
  - "in error" is confusing/ambiguous as there is no error message. It's not needed either, we can just say what was returned before and what it returned now.

Reply:
Done, and slightly simplified again